web/src/lib/utils.ts

```diff
@@ -308,8 +308,11 @@ export const findLocale = (code: string | undefined) => {
 };
 
 export const asyncTimeout = (ms: number) => {
-  return new Promise((resolve) => {
-    setTimeout(resolve, ms);
+  return new Promise<void>((resolve) => {
+    const timeout = setTimeout(() => {
+      clearTimeout(timeout);
```

Comment:
Calling `clearTimeout` is pointless because the timer has already fired by the time this callback is executed.

In the code you linked, the timeout can fire up to 5 seconds after leaving the page, but it won't make an API call. I think it's fine to leave it as is for simplicity's sake.